invictus_dashboard/src/app/shared/security/services/security.service.spec.ts

```diff
@@ -1,16 +1,74 @@
+import { HttpClient } from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
-
+import { RouterTestingModule } from '@angular/router/testing';
+import { SharedModule } from '../../shared.module';
+import { AppUser } from '../app-user';
+import { AppUserAuth } from '../app-user-auth';
+import { UsersData } from '../users-data';
 import { SecurityService } from './security.service';
 
 describe('SecurityService', () => {
   let service: SecurityService;
+  let httpClient: HttpClient;
+  let httpTestingController: HttpTestingController;
+  let testUserList: AppUser[];
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, SharedModule, RouterTestingModule],
+    });
+
+    httpClient = TestBed.inject(HttpClient);
+    httpTestingController = TestBed.inject(HttpTestingController);
     service = TestBed.inject(SecurityService);
+    testUserList = UsersData;
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should get inital list of users with a GET HTTP req', () => {
+    service.getUsers().subscribe({
+      next: (data: AppUser[]) => {
+        expect(data).toBe(testUserList);
```

Comment:
You want to be able to look at a test and have it be clear what the test data is and the expected output. When we take out the "arrange" data we lose that. I'd rather see a very small userlist object in the test than pulling this in from another file. That way we can see what the test is doing just by looking at the test. There is a pattern that is in some of the pluralsight path called Arrange Act Assert (AAA) that could help you with this.

Reply:
Got it, that makes sense. Replaced with a more visible list of users for clarity.